src/libs/ReportActionsUtils.js

```diff
@@ -76,7 +97,55 @@ function isConsecutiveActionMadeByPreviousActor(reportActions, actionIndex) {
     return currentAction.action.actorEmail === previousAction.action.actorEmail;
 }
 
+/**
+ * Get the message text for the last action that was not deleted
+ * @param {Number} reportID
+ * @param {Object} [actionsToMerge]
+ * @return {String}
+ */
+function getLastVisibleMessageText(reportID, actionsToMerge = {}) {
+    const parser = new ExpensiMark();
+    const actions = _.toArray(lodashMerge({}, allReportActions[reportID], actionsToMerge));
+    const sortedActions = _.sortBy(actions, 'sequenceNumber');
+    const lastMessageIndex = _.findLastIndex(sortedActions, action => (
+        !isDeletedAction(action)
+    ));
+    const htmlText = lodashGet(actions, [lastMessageIndex, 'message', 0, 'html'], '');
+    const messageText = parser.htmlToText(htmlText);
+    return ReportUtils.formatReportLastMessageText(messageText);
+}
+
+/**
+ * @param {Number} reportID
+ * @param {Object} [actionsToMerge]
+ * @param {Number} deletedSequenceNumber
+ * @param {Number} lastReadSequenceNumber
+ * @return {String}
+ */
+function getNewLastReadSequenceNumberForDeletedAction(reportID, actionsToMerge = {}, deletedSequenceNumber, lastReadSequenceNumber) {
+    // If the action we are deleting is unread then just return the current last read sequence number
+    if (deletedSequenceNumber > lastReadSequenceNumber) {
+        return lastReadSequenceNumber;
+    }
+
+    // Otherwise, we must find the first previous index of an action that is not deleted and less than the lastReadSequenceNumber
+    const actions = _.toArray(lodashMerge({}, allReportActions[reportID], actionsToMerge));
```

Comment:
NAB, we usually use `_.chain()` for these. It makes it more readable, but at the cost of debugging.